cmd/zq/zq.go

```diff
@@ -222,9 +236,18 @@ func (c *Command) Run(args []string) error {
 		if err != nil {
 			return err
 		}
-		for _, r := range readers {
+		for i, r := range readers {
 			if ndjr, ok := r.(*ndjsonio.Reader); ok {
-				if err := ndjr.SetTypeConfig(*tc); err != nil {
+				var path string
+				if c.jsonInferPath {
+					re := regexp.MustCompile(c.jsonPathRegexp)
+					match := re.FindStringSubmatch(paths[i])
+					if len(match) != 2 {
+						return fmt.Errorf("could not infer path from filename %s", paths[i])
+					}
+					path = match[1]
+				}
+				if err := ndjr.ConfigureTypes(*tc, path); err != nil {
```

Comment:
In an effort to keep the commands `Run` function simple I would move this logic into its own function (perhaps something like `configureJSONReader` and do the `ndjsonio.Reader` type coercion in the loop of the function `inputReaders`.

Reply:
👍